### DISCUSSION TITLE
What changed.

### DISCUSSION THREAD

**Discussion by ryanmurray** [nebhead/PiFire#94] [Q&A]
Updated to newest with a fresh os image and I hit start the temps and stuff don't display grill starts and lcd turns off. Uh ohhhhhh.

**Comment by ryanmurray** [comment]
And I did set my modules up in web GUi as soon as I hit start I see stop mode activated in events right after

**Comment by ryanmurray** [comment]
![Screenshot_20220412-194342.jpg](https://user-images.githubusercontent.com/1445752/163089589-11dccdbe-247e-412b-a87f-6476c5b1c619.jpg)

**Reply by weberbox** [reply]
What happens if you start it in the web ui? Does the web ui show any errors at the top? If a module is having issues loading etc there should now be a banner at the top that states an error has occurred 

We need to narrow it down to a server issue or Android issue.

**Reply by ryanmurray** [reply]
![Screenshot_20220412-211212_Brave.jpg](https://user-images.githubusercontent.com/1445752/163098677-7f56a4fc-0562-48ff-bac2-1cb8f650bbe2.jpg)

**Reply by ryanmurray** [reply]
And I'm doing monitor cause grill starts if I hit starts then says stop mode but I walk outside and grill is actually going 

My home screen does not show any module errors and I set them like I had them in 1.2.2 where I had everything working

**Reply by ryanmurray** [reply]
![Screenshot_20220412-211857_Brave.jpg](https://user-images.githubusercontent.com/1445752/163099305-82426011-25dd-4439-99bc-4a1fd30f789e.jpg)

**Reply by ryanmurray** [reply]
![Screenshot_20220412-213233_Brave.jpg](https://user-images.githubusercontent.com/1445752/163100691-cf91825b-46aa-4ef0-8947-37ea1b3a73af.jpg)

**Reply by ryanmurray** [reply]
Another weird thing I noticed upon boot the fan turns on and stays on maybe this is safety mode?

**Reply by weberbox** [reply]
The fan should not remain on all the time. It should run the fan during a mode cycle and when in shutdown. Normal operation has not really changed only new features were added like the config wizard and updater and some other features but basic operation should be just like before. 

Do you know what you selected for the relays? It would have been Active High or Active Low and the correct setting would depend on what type of relay board you have. 

Based on the pics above it looks like it is having trouble communicating with the VL53L0X. But that should usually trigger an error banner like the one below. So strange you are not seeing that. 

![image](https://user-images.githubusercontent.com/313328/163103453-b7e23e8b-a506-49fb-b7f2-867cb2a8d6b0.png)

Did you enable I2C after the new install in the raspi-config?

If you did enable it then it might be a good idea to take a look at the control.err.log and control.out.log in the /usr/local/bin/pifire/logs directory. 

Can you post the output or attach the files?

**Comment by ryanmurray** [comment]
I never got the red error about the distance sensor, i have changed it to prototype for the time being. I selected Active High which was what i had previously selected on 1.2.2. I2C and baud rate enable as i did prior install.

Control Error log has this repeating
]
KeyError: 'Grill1Temp'
Traceback (most recent call last):
  File "/usr/local/bin/pifire/control.py", line 1776, in <module>
    Monitor(grill_platform, adc_device, display_device, dist_device)
  File "/usr/local/bin/pifire/control.py", line 812, in Monitor
    adc_data = ReadProbes(settings, adc_device, units)
  File "/usr/local/bin/pifire/control.py", line 253, in ReadProbes
    prob_data[probe_ids[idx] + p] = adc_data[adc_probe_indices[source_index] + p]
KeyError: 'Grill1Temp'

Control log

Call of VL53L0X_DataInit
API Status: -20 : Control Interface Error
VL53L0X_BETTER_ACCURACY_MODE
Set Accuracy
API Status: -20 : Control Interface Error
2022-04-13 05:25:09 Error Reading Temperature.
Menu Active
Down pressed. Mode = inactive Value = 2 Selected = Stop
Down pressed. Mode = inactive Value = 1 Selected = Monitor
Enter pressed. Value = 1
Monitor Selected
2022-04-13 05:27:39 Error Reading Temperature.

I noticed if i manually go to monitor mode the screen shuts off and pifire logo reappears. Mind you this was working solid on 1.2.2 for months and i just pulled SD card with PI off and etched the latest 32 bit raspberry pi OS on, added WPA Supplicant, SSH and User Conf files and ran the script, Then enabled I2C locale etc., ran install script, set modules.

I noticed fan is not turning on by itself after disabling the distance sensor.

**Reply by nebhead** [reply]
> Mode = inactive Value = 2 Selected = Stop Down pressed. Mode = inactive Value = 1 Selected = Monitor Enter pressed. Value = 1 Monitor Selected 

Did you use the buttons to start monitor mode?  Or were these inputs erroneous?  As mentioned below, might be your button level is set wrong, but not sure.

**Reply by ryanmurray** [reply]
That last log was me using James button board with pullups and selecting monitor manually. As soon as I hit enter screen shut off

**Reply by nebhead** [reply]
OK, got it.  So button inputs are working fine.  It's something happening as soon as you start the work cycle.  I need to go back and look at your previous logs and see if there are any clues there.

**Comment by ryanmurray** [comment]
Seeing this upon boot also

dError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/pifire/control.py", line 38, in <module>
    ReadHistory(0, flushhistory=True)
  File "/usr/local/bin/pifire/common.py", line 779, in ReadHistory
    if cmdsts.exists('control:history'):
  File "/usr/local/lib/python3.9/dist-packages/redis/commands/core.py", line 1507, in exists
    return self.execute_command("EXISTS", *names)
  File "/usr/local/lib/python3.9/dist-packages/redis/client.py", line 1215, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
  File "/usr/local/lib/python3.9/dist-packages/redis/connection.py", line 1386, in get_connection
    connection.connect()
  File "/usr/local/lib/python3.9/dist-packages/redis/connection.py", line 620, in connect
    raise ConnectionError(self._error_message(e))
redis.exceptions.ConnectionError: Error 111 connecting to localhost:6379. Connection refused.

**Reply by nebhead** [reply]
> Seeing this upon boot also
> 
> dError: [Errno 111] Connection refused
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last): File "/usr/local/bin/pifire/control.py", line 38, in ReadHistory(0, flushhistory=True) File "/usr/local/bin/pifire/common.py", line 779, in ReadHistory if cmdsts.exists('control:history'): File "/usr/local/lib/python3.9/dist-packages/redis/commands/core.py", line 1507, in exists return self.execute_command("EXISTS", *names) File "/usr/local/lib/python3.9/dist-packages/redis/client.py", line 1215, in execute_command conn = self.connection or pool.get_connection(command_name, **options) File "/usr/local/lib/python3.9/dist-packages/redis/connection.py", line 1386, in get_connection connection.connect() File "/usr/local/lib/python3.9/dist-packages/redis/connection.py", line 620, in connect raise ConnectionError(self._error_message(e)) redis.exceptions.ConnectionError: Error 111 connecting to localhost:6379. Connection refused.

This error means that your Redis-Server isn't up and running for some reason.  It could be a race condition if it's happening during boot.  It seems like if you are able to use PiFire after boot-up then it has to be something else.  

Either way, your control script seems to be crashing for some reason.  It should be throwing an error banner on the WebUI, but I'm perplexed at what it could be.  One other thing to check is your Button Level setting.  When you went through the configuration, what did you set that to?  Previously I once had this set incorrectly and I had similar behavior to what you are seeing.

**Reply by ryanmurray** [reply]
I set to pullups which I believe is high and what Webber's button board uses

And I have not seen an error banner

**Reply by nebhead** [reply]
> I set to pullups which I believe is high and what Webber's button board uses
> 
> And I have not seen an error banner

Thanks.  We'll need to keep digging then.  

BTW - The above error that I mentioned the Redis-Server not being up and running seems to be consistent with a new install that I just made.  It's probably due to the fact that it takes time for the Redis-Server to get up and running and isn't quite ready when PiFire starts up.  I'll have to fix that somehow, but it's probably not the issue you are experiencing.  So the search continues.

**Reply by ryanmurray** [reply]
Morale of story don't upgrade the bbq right before a cook 🤣🤣🤣🤣

**Reply by nebhead** [reply]
> Morale of story don't upgrade the bbq right before a cook 🤣🤣🤣🤣

Ha!  Isn't this the truth.  Sigh.  We will definitely figure this out, don't you worry.  :)

**Reply by ryanmurray** [reply]
> > Morale of story don't upgrade the bbq right before a cook 🤣🤣🤣🤣
> 
> Ha!  Isn't this the truth.  Sigh.  We will definitely figure this out, don't you worry.  :) 

I can always burn another SD card with 1.2.2 and swap back and forth 😁

**Reply by nebhead** [reply]
I think we might want to focus on this error:

```
Traceback (most recent call last):
File "/usr/local/bin/pifire/control.py", line 1776, in
Monitor(grill_platform, adc_device, display_device, dist_device)
File "/usr/local/bin/pifire/control.py", line 812, in Monitor
adc_data = ReadProbes(settings, adc_device, units)
File "/usr/local/bin/pifire/control.py", line 253, in ReadProbes
prob_data[probe_ids[idx] + p] = adc_data[adc_probe_indices[source_index] + p]
KeyError: 'Grill1Temp'
```

It's looking for the Grill1Temp key in the dictionary an cannot find it.  The only reason this might happen is that the ADC module isn't returning this value.  Looking at the code in the ADC module, I have a bug, which I think you have exposed through having an issue with reading the I2C.  

Something is going wacky with your I2C, I believe.  I think you should back-out the I2C baud rate changes and see if that fixes anything.  Honestly, I am not using the baud rate increase for my Pi and it's working just fine.  You should be fine without that.  

In the meantime, I'm going to fix the bug that I found.  :)

**Reply by ryanmurray** [reply]
Okay I'll remove baudrate and just leave the i2c on. Let me know when push happens and I'll update my box

**Reply by nebhead** [reply]
Just pushed, so you can feel free to update whenever.  However, if my theory is correct, your this change would really matter too much.  It might just mask the I2C error.  Either way, curious to see how the baudrate change goes.  Might have to remove that from the documentation in the future.

**Reply by ryanmurray** [reply]
> Just pushed, so you can feel free to update whenever.  However, if my theory is correct, your this change would really matter too much.  It might just mask the I2C error.  Either way, curious to see how the baudrate change goes.  Might have to remove that from the documentation in the future. 

I2c baudrate removed and pulled updates and it's back to monitor mode working 😁

I'll test start when I'm home from work

**Comment by nebhead** [comment]
Great!  Are you getting actual temps in monitor mode?

**Reply by ryanmurray** [reply]
Yes I can successfully get temps on web mobile and app now under monitor mode. If I was home I'd hit start but don't want my wife calling me asking why the traeger is stuck on for hours LOL. Side note don't do updates with hungry wife right before dinner is what I should've said earlier she wasn't happy with my tinkering right when I got home 🤣🤣🤣🤣

**Reply by ryanmurray** [reply]
And how rude of me, thank you neb and Webber for the quick replies and fix!

**Reply by nebhead** [reply]
> Yes I can successfully get temps on web mobile and app now under monitor mode. If I was home I'd hit start but don't want my wife calling me asking why the traeger is stuck on for hours LOL. Side note don't do updates with hungry wife right before dinner is what I should've said earlier she wasn't happy with my tinkering right when I got home 🤣🤣🤣🤣

Ha!  Yes, I've learned my lesson too.  With anything in the Home Automation world, wait until you have plenty of free time without the Wife around to fiddle with stuff.  🤣

> And how rude of me, thank you neb and Webber for the quick replies and fix!

You are very welcome!  I'm pretty sure you'll be good when you get home today, but drop a comment here if you have any issues.

**Comment by ryanmurray** [comment]
Does same in webgui. Starts the immediately throws stop mode activated

**Reply by nebhead** [reply]
Hi @ryanmurray - Can you give some more details of your new issue?  Or maybe you can add a new issue in the issues area with details there?

**Reply by ryanmurray** [reply]
Hi ned I am really unsure how this became a thing 8 hours ago.... I've been rock solid for months and my GitHub app on my phone wasn't even signed in? 

IE I did not make a new post or comment

**Reply by nebhead** [reply]
Thanks!  I thought it was odd, out of the blue.  Must be a GitHub problem. :)  Glad to hear things are going well.  Take care!
